src/transformers/modeling_utils.py

```diff
@@ -147,6 +148,37 @@ def find_tensor_attributes(module: nn.Module) -> List[Tuple[str, Tensor]]:
         return first_tuple[1].dtype
 
 
+def get_parameter_first_float_dtype(parameter: Union[nn.Module, GenerationMixin, "ModuleUtilsMixin"]):
+    """returns the first found floating dtype in parameters or asserts if none were found"""
+    try:
+        for t in parameter.parameters():
+            if t.is_floating_point():
+                return t.dtype
+
+    except StopIteration:
+        # For nn.DataParallel compatibility in PyTorch 1.5
+
+        def find_tensor_attributes(module: nn.Module) -> List[Tuple[str, Tensor]]:
+            tuples = [(k, v) for k, v in module.__dict__.items() if torch.is_tensor(v)]
+            return tuples
+
+        gen = parameter._named_members(get_members_fn=find_tensor_attributes)
+        for tuple in gen:
+            if tuple[1].is_floating_point():
+                return tuple[1].dtype
+
```

Comment:
I'm not sure how we can handle this cleanly since the exception handler is already doing something else.

I proposed to return the last `tuple[1].dtype` if it falls through instead of the assert.

and I renamed the functions to indicate that they attempt to get float dtype or fallback to whatever dtype is found.